notebooks/airline_passenger_satisfaction_predictor.ipynb

```diff
@@ -1541,26 +1549,221 @@
+       "Logistic Regression  0.817 (+/- 0.122)  0.094 (+/- 0.009)   0.934 (+/- 0.002)   \n",
+       "Decision Tree        2.331 (+/- 0.018)  0.092 (+/- 0.003)   0.947 (+/- 0.001)   \n",
+       "\n",
+       "                        train_accuracy  \n",
+       "dummy                0.567 (+/- 0.000)  \n",
+       "Logistic Regression  0.934 (+/- 0.000)  \n",
+       "Decision Tree        1.000 (+/- 0.000)  "
+      ]
+     },
+     "execution_count": 82,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
    "source": [
     "results_df"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "id": "ab740e95-179e-41ae-aa97-200a9f68dcd6",
+   "metadata": {},
+   "source": [
+    "Based on the result, we can see that the Decision Tree model has the highest validation accuracy at 94.7%. Its training accuracy is 100%, which indicates it perfectly fits the training data but may be overfitting. The validation accuracy for Logistic Regression is 93.4%, slightly lower than the Decision Treem, and the training accuracy is close to its validation accuracy, suggesting it generalizes better and is less likely to overfit. So we would use Logistic Regression as our final model on predicting the test set."
+   ]
+  },
   {
    "cell_type": "code",
-   "execution_count": null,
-   "id": "a2e0ac13-c497-4487-82cd-d4c1e492416c",
+   "execution_count": 83,
+   "id": "fe436852-287e-4103-894f-0c3256cc5f1e",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Logistic Regression Test Accuracy: 0.9322836464428703\n",
+      "\n",
+      "Classification Report:\n",
+      "                          precision    recall  f1-score   support\n",
+      "\n",
+      "neutral or dissatisfied       0.93      0.95      0.94     14573\n",
+      "              satisfied       0.93      0.91      0.92     11403\n",
+      "\n",
+      "               accuracy                           0.93     25976\n",
+      "              macro avg       0.93      0.93      0.93     25976\n",
+      "           weighted avg       0.93      0.93      0.93     25976\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Prepare the test set\n",
+    "X_test = test_data.drop(columns=['satisfaction'])\n",
+    "y_test = test_data['satisfaction']\n",
+    "y_test = y_test.values.ravel()\n",
+    "\n",
+    "# Define the Logistic Regression pipeline\n",
+    "final_model = make_pipeline(preprocessor, LogisticRegression(random_state=123))\n",
+    "\n",
+    "# Train the model on the training data\n",
+    "final_model.fit(X_train, y_train.values.ravel())  \n",
+    "\n",
+    "# Predict on the test set\n",
+    "y_test_pred = final_model.predict(X_test)\n",
+    "\n",
+    "# Evaluate the model performance\n",
+    "print(\"Logistic Regression Test Accuracy:\", accuracy_score(y_test, y_test_pred))\n",
+    "print(\"\\nClassification Report:\\n\", classification_report(y_test, y_test_pred))"
```

Comment:
Where are the functions `accuracy_score` and `classification_report` defined/imported